### DISCUSSION TITLE
Running inference using CUDAExecutionProvider without cuDNN

### DISCUSSION THREAD

**Discussion by devdrian** [microsoft/onnxruntime#18083] [Other Q&A] [answered]
Hi all,

In Python, using onnxruntime-gpu, is it possible to use the CUDAExecutionProvider without having the cuDNN library available? I try to not include cuDNN in our project, as it significantly increases loading times.

I already tried setting different SessionOptions and options to the CUDAExecutionProvider, e.g., disabling optimizations performed by cuDNN, but that seems to be not enough.

Does anyone have an idea or knows if that's possible at all? Maybe something similar to PyTorch, where there is a compilation flag to disable cuDNN?

**Comment by RyanUnderhill** [comment, marked as answer]
No, the CUDAExecutionProvider uses cuDNN for some of the kernels it supports. So there's no way to build the CUDA provider without it:

https://onnxruntime.ai/docs/build/eps.html#cuda

**Reply by devdrian** [reply]
Thanks a lot for your answer, Ryan!
